runtime/vm/ContinuationHelpers.cpp

```diff
@@ -261,7 +265,8 @@ freeContinuation(J9VMThread *currentThread, j9object_t continuationObject)
 			currentStack = previous;
 		} while (NULL != currentStack);
 
-		Assert_VM_true(VM_VMHelpers::isFinished(continuation->state));
+		ContinuationState *continuationStatePtr = VM_VMHelpers::getContinuationStateAddress(currentThread, continuationObject);
```

Comment:
I'd fetch the state, rather then the address:

ContinuationState continuationState = *VM_VMHelpers::getContinuationStateAddress(currentThread, continuationObject);